src/Courses/CourseDetailsTabs/StatisticsTab.tsx

```diff
@@ -222,15 +225,19 @@ export const StatisticsTab: React.FC<StatisticsTabProps> = ({ course, userId })
                         completionPercent: formatNumberString(d.completionPercent, true)
                     }));
                 }
+                logger.silly('Stats tab: [useEffect] setting rowData');
                 setRowData(data);
+                setLoading(false);
             } catch (e) {
                 logger.error('Failed to get statistics.', e);
+                setLoading(false);
```

Comment:
We could remove the below return and put the setLoading outside the try catch

Better yet we could put setLoading in a finally block

Reply:
done